sktime/forecasting/base/adapters/_neuralforecast.py

```diff
@@ -189,15 +191,47 @@ def _fit(
         if not fh.is_all_out_of_sample(cutoff=self.cutoff):
             raise NotImplementedError("in-sample prediction is currently not supported")
 
-        if self.freq == "auto" and fh.freq is None:
-            # when freq cannot be interpreted from ForecastingHorizon
-            raise ValueError(
-                f"Error in {self.__class__.__name__}, "
-                f"could not interpret freq, "
-                f"try passing freq in model initialization"
-            )
-
-        self._freq = fh.freq if self.freq == "auto" else self.freq
+        # A. freq is given {use this}
+        # B. freq is auto
+        #     B1. freq is infered from fh {use this}
+        #     B2. freq is not infered from fh
+        #         B2.1. y is date-like {raise exception}
+        #         B2.2. y is not date-like
+        #             B2.2.1 equispaced integers {use diff in time}
+        #             B2.2.2 non-equispaced integers {raise exception}
+
+        if self.freq != "auto":
+            # A
+            self._freq = self.freq
+        else:
+            # B
+            if fh.freq:
+                # B1
+                self._freq = fh.freq
+            else:
+                # B2
+                if isinstance(y.index, (pandas.PeriodIndex, pandas.DatetimeIndex)):
+                    # B2.1
+                    raise ValueError(
+                        f"Error in {self.__class__.__name__}, "
+                        f"could not interpret freq, "
+                        f"try passing freq in model initialization "
+                        f"or use a valid offset in index"
```

Comment:
```suggestion
                        f"could not interpret freq ({freq}), "
                        "try passing freq in model initialization "
                        "or use a valid offset in index"
```

Reply:
@VascoSch92 does `freq` object exist at this point?

My understanding was this error will be raised only when user has passed `freq="auto"` and it's up to `fh` to interpret the frequency which fails.

Reply:
`freq` object doesnot exist here